src/Environment.php

```diff
@@ -0,0 +1,99 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\Markdown;
+
+final class Environment<T> {
+  public function __construct(
+    private ParserContext $parser,
+    private RenderContext $context,
+    private Renderer<T> $renderer,
+  ) {}
+
+  public static function html(bool $unsafe = false): Environment<string> {
+    $parser = new ParserContext();
+    $context = new RenderContext();
+    $renderer = new HTMLRenderer($context);
+
+    if ($unsafe) {
+        $parser->enableHTML_UNSAFE();
+    }
+
+    return new self($parser, $context, $renderer);
+  }
+
+  public function setParser(ParserContext $parser): void {
```

Comment:
Do we need these getters and setters? I would definitely remove the setters at least (we don't want someone to change the context in the middle of rendering, or really at any point after it's assigned in the constructor).